### DISCUSSION TITLE
Understanding multiple URLs when setting up the RavenDB client

### DISCUSSION THREAD

**Discussion by dedels** [ravendb/ravendb#15217] [General]
https://ravendb.net/docs/article-page/5.4/csharp/client-api/creating-document-store
When I setup my document store connection it asks for multiple URLs.  I have a 3 node cluster so I will put in the 3 nodes I know of.  

What does this do?  If one node goes down, I would guess my client API will try the next one (HA).  
Which node do I connect to by default here?  the first in my list?  If I have a cluster of web apps, should they randomize the URL order?  Or the client API will select different nodes?

If my Raven cluster is more than 3 nodes... is it ok to still connect to the 3?  will the client API discover more nodes?

**Comment by georgiosd** [comment]
The client API will manage that for you internally so I wouldn't mess with it.

It will pick one node to write to, until it fails and then go to another one. Unless `FastestNode` is enabled for reads, which will use the fastest node for reads... 

Not sure about the discovery of more nodes but curious about what use case you have that warrants more than 3 nodes!

**Comment by ayende** [comment]
In general, that is a pretty complex behavior.

We use those URLs solely for discovery, nothing else. The client will ping each one of those nodes in order and ask it for the _actual_ URLs for the database in question.

Multiple nodes will use the same topology (which may be updated on the fly, etc).

There is _no_ meaning to the order of the URLs between clients.
